libpromises/attributes.c

```diff
@@ -364,6 +364,7 @@ ExecContain GetExecContainConstraints(const EvalContext *ctx, const Promise *pp)
     e.owner = PromiseGetConstraintAsUid(ctx, "exec_owner", pp);
     e.group = PromiseGetConstraintAsGid(ctx, "exec_group", pp);
     e.preview = PromiseGetConstraintAsBoolean(ctx, "preview", pp);
+    e.have_nooutput = PromiseBundleOrBodyConstraintExists(ctx, "no_output", pp);
```

Comment:
Oh, you're right. I thought originally it wasn't possible because this is in the body, and what we would need to check would be in the surrounding promise. But it looks like both may be available here. I'll check.